assets/src/dashboard/app/views/myStories/content/storyGridItem/index.js

```diff
@@ -0,0 +1,149 @@
+  StoryMenu,
+} from '../../../../../components';
+import { generateStoryMenu } from '../../../../../components/popoverMenu/story-menu-generator';
+import { STORY_STATUS } from '../../../../../constants';
+import {
+  PageSizePropType,
+  RenameStoryPropType,
+  StoryMenuPropType,
+  StoryPropType,
+} from '../../../../../types';
+
+const CustomCardGridItem = styled(CardGridItem)`
+  display: grid;
+  grid-template-columns: 100%;
+  grid-template-rows: ${({ pageSize }) => `${pageSize.posterHeight}px auto`};
+`;
+
+export const DetailRow = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+`;
+
+const StoryGridItem = ({
+  bottomActionLabel,
+  handleFocus,
+  isActive,
+  itemRefs = {},
+  pageSize,
+  renameStory,
+  story,
+  storyMenu,
+}) => {
+  const tabIndex = isActive ? 0 : -1;
+  const titleRenameProps = renameStory
+    ? {
+        editMode: renameStory?.id === story?.id,
+        onEditComplete: (newTitle) =>
+          renameStory?.handleOnRenameStory(story, newTitle),
+        onEditCancel: renameStory?.handleCancelRename,
+      }
+    : {};
+
+  const generatedMenuItems = useMemo(
+    () =>
+      generateStoryMenu({
+        menuItemActions: storyMenu.menuItemActions,
+        menuItems: storyMenu.menuItems,
+        story,
+      }),
+    [storyMenu, story]
+  );
+
+  const storyDate = getRelativeDisplayDate(
+    story?.status === STORY_STATUS.DRAFT
+      ? story?.modified_gmt
+      : story?.created_gmt
+  );
+  return (
+    <CustomCardGridItem
+      data-testid={`story-grid-item-${story.id}`}
+      ref={(el) => {
+        itemRefs.current[story.id] = el;
+      }}
+      onFocus={handleFocus}
+      title={sprintf(
+        /* translators: %s: story title.*/
+        __('Details about %s', 'web-stories'),
+        story.title
+      )}
+      pageSize={pageSize}
+    >
+      <StoryCardPreview
+        tabIndex={tabIndex}
+        storyImage={story.featuredMediaUrl}
+        storyTitle={story.title}
+        bottomAction={{
+          targetAction: story.bottomTargetAction,
+          label: bottomActionLabel,
+        }}
+      />
+      <DetailRow>
+        <CardTitle
+          tabIndex={tabIndex}
+          title={story.title}
+          titleLink={story.editStoryLink}
+          status={story?.status}
+          locked={story?.locked}
+          lockUser={story?.lockUser}
+          id={story.id}
+          displayDate={storyDate}
+          {...titleRenameProps}
+        />
+
+        <StoryMenu
+          itemActive={isActive}
+          tabIndex={tabIndex}
+          onMoreButtonSelected={storyMenu.handleMenuToggle}
+          contextMenuId={storyMenu.contextMenuId}
+          storyId={story.id}
+          menuItems={generatedMenuItems}
+        />
+      </DetailRow>
+    </CustomCardGridItem>
+  );
+};
+
+StoryGridItem.propTypes = {
+  bottomActionLabel: PropTypes.string,
+  handleFocus: PropTypes.func,
+  isActive: PropTypes.bool,
+  itemRefs: PropTypes.shape({}),
```

Comment:
💭 Maybe we could make this type a little more specific since we know this will be a reference to an element type?
```suggestion
  itemRefs: PropTypes.shape({
    current: PropTypes.oneOfType([
        PropTypes.func, 
        PropTypes.shape({ current: PropTypes.instanceOf(Element) })
      ])
  }),
```

Reply:
thank you great idea!